x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -30,8 +31,14 @@ export function initAuthenticateApi(server) {
       const { username, password } = request.payload;
 
       try {
-        request.loginAttempt().setCredentials(username, password);
-        const authenticationResult = await server.plugins.security.authenticate(request);
+        // We should prefer `token` over `basic` if possible.
+        const providerToLoginWith = config.authc.providers.includes('token')
```

Comment:
__note:__ not sure how I feel about this, can't think of a better alternative right now :man_shrugging:  This choice was implicitly made by authenticator before, now it's explicitly made here.